lib/generators/active_record/devise_security_generator.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require 'rails/generators/active_record'
+require 'generators/devise/orm_helpers'
+
+module ActiveRecord
+  module Generators
+    class DeviseSecurityGenerator < Base
+      source_root File.expand_path('templates', __dir__)
+
+      def copy_migration_file
+        migration_template 'session_histories.rb.erb',
+                           "db/migrate/create_#{table_name}.rb",
+                           migration_version: migration_version,
+                           ip_column: ip_column
+      end
+
+      def ip_column
+        inet? ? 'inet' : 'string'
+      end
+
+      def inet?
+        postgresql?
+      end
+
+      def postgresql?
+        config = ActiveRecord::Base.configurations[Rails.env]
+        config && config['adapter'] == 'postgresql'
+      end
+
+      def migration_version
+        "[#{Rails::VERSION::MAJOR}.#{Rails::VERSION::MINOR}]" if Rails.version.start_with?('5')
```

Comment:
Why the ` if Rails.version.start_with?('5')`? I think this is going to be the case going forward for 6+

Reply:
Yeah, this should account for future rails versions as well. At least 5 & 6 for now.

Reply:
Since it's still supporting rails 4.2 based on the gems dependency. We can drop rails 4.2 support since it's already [EOF](https://rubyonrails.org/maintenance/).

Reply:
Updated it to be more compatible with rails 5 & up.